src/structarray.jl

```diff
@@ -453,6 +453,22 @@ function StructArrayStyle{S, M}(::Val{N}) where {S, M, N}
     return StructArrayStyle{T, N}()
 end
 
+_dimmax(a::Integer, b::Integer) = max(a, b)
+_dimmax(::Type{Any}, ::Integer) = Any
+_dimmax(::Integer ,::Type{Any}) = Any
+_dimmax(::Type{Any} ,::Type{Any}) = Any
```

Comment:
This function is a bit mysterious outside of the context where it is used. Maybe best to just add something like

```julia
N′ = M === Any || N === Any ? Any : max(M, N)
```

below.